cpp/src/arrow/device.h

```diff
@@ -165,6 +198,17 @@ class ARROW_EXPORT MemoryManager : public std::enable_shared_from_this<MemoryMan
   static Result<std::shared_ptr<Buffer>> ViewBuffer(
       const std::shared_ptr<Buffer>& source, const std::shared_ptr<MemoryManager>& to);
 
+  /// \brief Create a SyncEvent for exporting
+  ///
+  /// This version should construct the appropriate event for the device and
+  /// provide the unique_ptr with the correct deleter for the event type.  
+  virtual Result<std::shared_ptr<Device::SyncEvent>> MakeDeviceSync();
+
+  /// \brief Create a SyncEvent from imported device array.
+  ///   
+  /// @param sync_event passed in sync_event from the imported device array.
+  virtual Result<std::shared_ptr<Device::SyncEvent>> MakeDeviceSync(std::unique_ptr<void, void(*)(void*)> sync_event);
```

Comment:
Can we name these as `MakeDeviceSyncEvent`? I read these as "Instruct the device to synchronize" 😅